### ISSUE TITLE
TableDefinition does not support sql server identity columns of type BigInt

### ISSUE THREAD

**Issue by LarsAlnor** [roadrunnerlenny/etlbox#40]
**Describe the bug**
The ReadTableDefinitionFromSqlServer has two casts (of seed_value and increment_value) to int? You get an invalid cast exception, if your identity column is of type BigInt. You can't cast from long to int?.

**Steps To Reproduce**
Identity column of table is of type BigInt

**Actual behaviour**
Invalid cast exception.

**Expected behavior**
No exception as it is a valid table definition.

One way to solve it is to use Convert:
            , seed_value => curCol.IdentitySeed = (int?)(Convert.ToInt32(seed_value))
            , increment_value => curCol.IdentityIncrement = (int?)(Convert.ToInt32(increment_value))

**Comment by roadrunnerlenny** [comment]
Hi,

thanks for using ETLBox and reporting the bug! Looks like it should be easy to fix, I will look into it (and add also some tests). I will  release a new version containing the fix hopefully in the next couple of days.

In the meanwhile, there is a workaround to avoid having the the TableDefiniton read from the database: In the `DBSource`, you simple pass your own `TableDefinition` object - and define here what column actually is the identity column:

```C#
var TableDefinition = new TableDefinition("table1"
                , new List<TableColumn>() {
                new TableColumn("Id", "BIGINT", allowNulls:false,  isPrimaryKey: true, isIdentity:true)),
                new TableColumn("OtherCol", "NVARCHAR(100", allowNulls: true)
            });

var DBSource<type> = new DBSource<type>() {  
  SourceTableDefinition = TableDefinition
}
```

**Comment by roadrunnerlenny** [comment]
Hi, 
I just released version 1.6.5 which fixes the bug. Now you should be able to create a `DBSource` or to call the `ReadTableDefinitionFromSqlServer` method on a table that has a BIGINT identity column.
